### DISCUSSION TITLE
Screening in batches

### DISCUSSION THREAD

**Discussion by n400peanuts** [asreview/asreview#1781] [Q&A] [answered]
Hello,

I have a collaborator (A) that is able to screen papers only if I provide them already tabled in excel (so no installing asreview locally). This is the only person screening papers.

I was hoping that I (B) would provide asreview with the feedback from A (let's say, 50 papers at a time), then re-train the model with the updated knowledge and get a new ranking. So essentially this would be the plan:

day 0. 

Set up asreview, provide prior knowledge (e.g., 11 relevant, 5 irrelevant), start reviewing. Get the initial ranking based on the prior knowledge, export it to excel, send it to A. 

day 1. 
A provides feedback on the first 50 ranked papers, gives back to B the labels (1/0 relevant/irrelevant), then B provides the updated feedback back to asreview to re-do the ranking with the novel feedback.

day 2.
Cycle repeats.

Do you think this would be possible? and how would I approach it? I have exported already the output of day 0, and A provided feedback on 50 papers, how do I combine now this knowledge back to asreview? should I start a new project?

**Comment by Rensvandeschoot** [comment, marked as answer]
This is a fascinating approach! It should work if you create a new project after every screening day of A. As our paper on [reproducibility](https://doi.org/10.3390/app14093842) explains, the model takes all labeled records and trains a new model from scratch every iteration. An iteration can involve one screened paper, or, as in your case, an iteration can involve all newly labeled papers after a day of screening. In both cases, the newly labeled records (1 or many) are combined with all labeled records from previous iterations plus the prior knowledge.

**Reply by n400peanuts** [reply]
Thank you for your swift reply! 
indeed creating a new project does the trick, exactly as you described. 

However, while I was trying to get the latest ranking and add the metadata of the papers, it took me a moment to understand that the `record_id` of the output of the second iteration (round2) exported via this code (found in the documentation):

```python
with open_state("bib/round2/output/pubmed-9NOTanimal-set 2-ROUND2.asreview") as state:
     last_ranking = state.get_last_ranking()
```

Output:

```python
last_ranking.head()
```

| record_id | ranking | classifier | query_strategy  | balance_strategy | feature_extraction | training_set | time                       |
|-----------|---------|------------|-----------------|------------------|--------------------|--------------|---------------------------|
| 0         | 2       | 0          | nb              | max_uncertainty  | double             | tfidf        | 53           | 2024-07-17 14:20:34.895346 |
| 1         | 0       | 1          | nb              | max_uncertainty  | double             | tfidf        | 53           | 2024-07-17 14:20:34.895346 |
| 2         | 14      | 2          | nb              | max_uncertainty  | double             | tfidf        | 53           | 2024-07-17 14:20:34.895346 |
| 3         | 9       | 3          | nb              | max_uncertainty  | double             | tfidf        | 53           | 2024-07-17 14:20:34.895346 |
| 4         | 17      | 4          | nb              | max_uncertainty  | double             | tfidf        | 53           | 2024-07-17 14:20:34.895346 |

is the index number of the input file, which is the same as the `record_id` and `ranking` column of the input file. So when one wants to retrieve all the information regarding the papers (e.g., titles, authors, abstracts, etc.) contained in the input file and merge it with the output of round2, it needs to be merged via the `record_id` column. However, this latter column will not reflect (in the case of the 2nd round) the ranking number (like in the input file). So, this is important to know—I got confused very easily, and became afraid that `record_id` of the output of the 2nd round was reflecting the current index of this latest dataframe, and I could not therefore retrieve the info re- papers. Does this make sense? I realise that I spent an entire day on this tiny detail and I might sound too specific... However this precision becomes critical when one goes for round 3, as this means that I should load the input of round 2, and merge by `record_id`.

Perhaps this was obvious to you and to the whole community, and the point is that ASReview was not built to do screening in this way (i.e., in batches), and so this confusion does not happen very often (nobody merges the output of a round of review with the previous one to see differences in ranking and extract metadata).

Related to this point, I wonder if there is a function already in the ASReview toolset that does this merge for me, so that it is less error-prone. Essentially, a function that extracts from `state` (via `open_state()`) a dataframe that contains all the info regarding the ranking of the papers from the last iteration (that is the output of `get_last_ranking()`) + the metadata related to the `record_id` (that is authors, title, abstracts) and perhaps a column `already_seen` that indicates which ones have been seen / given feedback already (though I guess this could be inferred from the `Included` column so this is not crucial)

**Reply by Rensvandeschoot** [reply]
Would the `compose` functionality of [ASReview-datatools](https://github.com/asreview/asreview-datatools) work for your situation? 

The record_id variable is for internal use and is overwritten, right @J535D165?
